### PR TITLE
Add `example_datasets.py` DAG to unit tests

### PR DESCRIPTION
**Pull request by phanikumv**
closes: #816

### CODE REVIEW

**Comment by kaxil** [comment]
Markdown link check should be fixed by https://github.com/astronomer/astro-sdk/pull/842

File: python-sdk/tests/test_example_dags.py [hunk 1]
```diff
@@ -93,3 +93,22 @@ def test_example_dynamic_task_map_dag(session, dag_id):
...
+    ],
+)
+def test_example_dataset_dag(session, dag_id):
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    db = DagBag(dir_path + "/../example_dags")
+    dag = db.get_dag(dag_id)
+
+    if dag is None:
+        raise NameError(f"The DAG with dag_id: {dag_id} was not found")
+    wrapper_run_dag(dag)
```
Comment (kaxil) [inline python-sdk/tests/test_example_dags.py:114]: Did you test it with 2.4.0b1 .. you can test it out on CI by bumping airflow dep to >=2.4.0b1 

I have a feeling this won't work as the second Dataset wont be triggered by the logic in `wrapper_run_dag`

Comment (kaxil) [reply python-sdk/tests/test_example_dags.py:114]: Bump the following to 2.4.0b1
https://github.com/astronomer/astro-sdk/blob/330c02a4d455759f9a164dde253a32747a7d2f5f/python-sdk/pyproject.toml#L20

Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py:114]: No I havent tested with 2.4.0b1, doing it now

File: python-sdk/tests/test_example_dags.py [hunk 2]
```diff
@@ -92,3 +92,22 @@ def test_example_dynamic_task_map_dag(session, dag_id):
...
     wrapper_run_dag(dag)
+
+
+@pytest.mark.skipif(
+    airflow.__version__ < "2.4.0", reason="Require Airflow version >= 2.4.0"
+)
+@pytest.mark.parametrize(
+    "dag_id",
+    [
+        "example_dataset_producer",
```
Comment (kaxil) [inline python-sdk/tests/test_example_dags.py:103]: Do we need to test that consumer DAG is triggered ?

File: python-sdk/tests/test_example_dags.py (hunk 1 above)
Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py:113]: somehow CI doesnt seem to run tests on 2.4.0b1

File: python-sdk/tests/test_example_dags.py (hunk 2 above)
Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py:103]: Yes , and this will need >=2.4.0

File: python-sdk/tests/test_example_dags.py (hunk 1 above)
Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py:113]: It has picked up the DAG successfully
`5.85s call     tests/test_example_dags.py::test_example_dataset_dag[example_dataset_producer]`

File: python-sdk/tests/test_example_dags.py (hunk 2 above)
Comment (utkarsharma2) [reply python-sdk/tests/test_example_dags.py:103]: I think we can skip testing this since it's something specific to airflow and not AstroSDK.

Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py:103]: Just wondering how can we ensure that if a developer changes the code relevant to mapping the Datasets to Astro-SDK, that it is not breaking any functionality?

File: python-sdk/tests/test_example_dags.py (hunk 1 above)
Comment (phanikumv) [reply python-sdk/tests/test_example_dags.py]: As discussed, moved the test to `python-sdk/tests/airflow/test_datasets.py` . Its working fine with `2.4.0b1`

File: python-sdk/tests/airflow/test_datasets.py [hunk 3]
```diff
@@ -76,3 +77,26 @@ def test_kwargs_with_datasets(
...
         )
+
+
+@pytest.mark.skipif(
+    airflow.__version__ < "2.4.0", reason="Require Airflow version >= 2.4.0"
+)
+def test_example_dataset_dag():
+    import os
+
+    from airflow.models.dagbag import DagBag
```
Comment (kaxil) [inline python-sdk/tests/airflow/test_datasets.py:88]: This two can go at the top

**Review by kaxil — APPROVED** [review]

**Pull request merged**